File: React_DnD/react-dnd/src/domain/character/value-object/CharacterProfile.js
// C:\Users\Laptop-JAB\Desktop\Learn\React_DnD\react-dnd\src\domain\character\value-object\CharacterProfile.js

export class CharacterProfile {
  constructor({
    age = 0,
    height = "",
    weight = "",
    eyes = "",
    skin = "",
    hair = "",
  } = {}) {
    this.age = this.validateAge(age);
    this.height = height;
    this.weight = weight;
    this.eyes = eyes;
    this.skin = skin;
    this.hair = hair;

    Object.freeze(this);
  }

  validateAge(age) {
    if (typeof age !== "number" || age < 0) return 0;
    return age;
  }

  toJSON() {
    return {
      age: this.age,
      height: this.height,
      weight: this.weight,
      eyes: this.eyes,
      skin: this.skin,
      hair: this.hair,
    };
  }
}

File: dnd-full-version/DndGame.Api/Entities/CharacterProfile.cs
namespace DndGame.Api.Entities;

public class CharacterProfile
{
    public int Age { get; }
    public string Height { get; }
    public string Weight { get; }
    public string Eyes { get; }
    public string Skin { get; }
    public string Hair { get; }

    public CharacterProfile(
        int age = 0,
        string height = "",
        string weight = "",
        string eyes = "",
        string skin = "",
        string hair = ""
    )
    {
        Age = ValidateAge(age);
        Height = height;
        Weight = weight;
        Eyes = eyes;
        Skin = skin;
        Hair = hair;
    }

    private int ValidateAge(int age)
    {
        if (age < 0)
            return 0;

        return age;
    }
}